qiskit_experiments/calibration_management/base_calibrations.py

```diff
@@ -0,0 +1,148 @@
+                generated after the cutoff date will be ignored. If the cutoff_date is None then
+                all parameters are considered. This allows users to discard more recent values that
+                may be erroneous.
+
+        Returns:
+            value: The value of the parameter.
+        """
+
+    @abstractmethod
+    def get_schedule(
+        self,
+        name: str,
+        qubits: Union[int, Tuple[int, ...]],
+        assign_params: Dict[Union[str, ParameterKey], ParameterValueType] = None,
+        group: Optional[str] = "default",
+        cutoff_date: datetime = None,
+    ) -> ScheduleBlock:
+        """Get the template schedule with parameters assigned to values.
+
+        All the parameters in the template schedule block will be assigned to the values managed
+        by the calibrations unless they are specified in assign_params. In this case the value in
+        assign_params will override the value stored by the calibrations. A parameter value in
+        assign_params may also be a :class:`ParameterExpression`.
+
+        .. code-block:: python
+
+            # Get an xp schedule with a parametric amplitude
+            sched = cals.get_schedule("xp", 3, assign_params={"amp": Parameter("amp")})
+
+            # Get an echoed-cross-resonance schedule between qubits (0, 2) where the xp echo gates
+            # are referenced schedules but leave their amplitudes as parameters.
+            assign_dict = {("amp", (0,), "xp"): Parameter("my_amp")}
+            sched = cals.get_schedule("cr", (0, 2), assign_params=assign_dict)
+
+        Args:
+            name: The name of the schedule to get.
+            qubits: The qubits for which to get the schedule.
+            assign_params: The parameters to assign manually. Each parameter is specified by a
+                ParameterKey which is a named tuple of the form (parameter name, qubits,
+                schedule name). Each entry in assign_params can also be a string corresponding
+                to the name of the parameter. In this case, the schedule name and qubits of the
+                corresponding ParameterKey will be the name and qubits given as arguments to
+                get_schedule.
+            group: The calibration group from which to draw the parameters. If not specified
+                this defaults to the 'default' group.
+            cutoff_date: Retrieve the most recent parameter up until the cutoff date. Parameters
+                generated after the cutoff date will be ignored. If the cutoff_date is None then
+                all parameters are considered. This allows users to discard more recent values that
+                may be erroneous.
+
+        Returns:
+            schedule: A copy of the template schedule with all parameters assigned.
+        """
+
+    @abstractmethod
+    def has_template(self, schedule_name: str, qubits: Optional[Tuple[int, ...]] = None) -> bool:
```

Comment:
It is in the base class because a calibration experiment needs it. See here: https://github.com/Qiskit-Extensions/qiskit-experiments/blob/a387675a3fe817cef05b968bbf3e05799a09aaae/qiskit_experiments/library/calibration/rough_amplitude_cal.py#L288

Reply:
Why we cannot just do
```python
try:
    schedule = self._cals.get_schedule("x", self.physical_qubits)
except CalibrationError:
    pass
```
?

Reply:
I think that's fine. I remove `has_template` from the base class. See here: [6f21049](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1200/commits/6f210497fda4f690ad0ff8b4dc58db22e2501f5a)